launcher/SysInfo.cpp

```diff
@@ -0,0 +1,64 @@
+#include <QString>
+#ifdef Q_OS_MACOS
+#include <sys/sysctl.h>
+#endif
+
+#ifdef Q_OS_MACOS
+bool rosettaDetect() {
+    int ret = 0;
+    size_t size = sizeof(ret);
+    if (sysctlbyname("sysctl.proc_translated", &ret, &size, NULL, 0) == -1)
+    {
+        return false;
+    }
+    if(ret == 0)
+    {
+        return false;
+    }
+    if(ret == 1)
+    {
+        return true;
+    }
+    return false;
+}
+#endif
+
+namespace SysInfo {
+    QString currentSystem() {
+#ifdef Q_OS_LINUX
+        return "linux";
+#elif Q_OS_MACOS
+        return "osx";
+#elif Q_OS_WINDOWS
+        return "windows";
+#elif Q_OS_FREEBSD
+        return "freebsd";
+#elif Q_OS_OPENBSD
+        return "openbsd";
+#else
+        return "unknown";
+#endif
+    };
+
+    QString currentArch() {
+#ifdef Q_PROCESSOR_X86_64
+    #ifdef Q_OS_MACOS
+        if(rosettaDetect())
+        {
+          return "arm64";
+        }
+        else
+        {
+          return "amd64";
+        }
+    #else
+        return "amd64";
```

Comment:
This doesn't consider systems where multiple architectures work. For example if soneone is running a 32-bit build on 64-bit Windows. This could cause issues if there are split natives for x86 and x86-64

Reply:
Oh yeah. I'll try to figure out that then. Probably abiInfo will be good